src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseAnalyzer.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Meziantou.Analyzer.Rules
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AvoidUsingRedundantElseAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_rule = new DiagnosticDescriptor(
+            RuleIdentifiers.AvoidUsingRedundantElse,
+            title: "Avoid using redundant 'else'",
+            messageFormat: "Avoid using redundant 'else'",
+            RuleCategories.Style,
+            DiagnosticSeverity.Info,
+            isEnabledByDefault: true,
+            description: "The 'if' block contains a jump statement (break, continue, goto, return, throw, yield break). Using 'else' is redundant and needlessly maintains a higher nesting level.",
+            helpLinkUri: RuleIdentifiers.GetHelpUri(RuleIdentifiers.AvoidUsingRedundantElse));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(AnalyzeElseClause, SyntaxKind.ElseClause);
+        }
+
+        private static void AnalyzeElseClause(SyntaxNodeAnalysisContext context)
+        {
+            var elseClauseSyntax = (ElseClauseSyntax)context.Node;
+            if (elseClauseSyntax is null)
+                return;
+
+            if (!(elseClauseSyntax.Parent is IfStatementSyntax ifStatementSyntax))
+                return;
+
+            var statementOrBlock = ifStatementSyntax.Statement;
+            if (statementOrBlock is null)
+                return;
+
+            var result = context.SemanticModel.AnalyzeControlFlow(statementOrBlock);
+            if (!result.Succeeded)
+                return;
+
+            if (!result.EndPointIsReachable || result.ExitPoints.Any(ep => IsDirectAccess(ifStatementSyntax, ep)))
+            {
+                context.ReportDiagnostic(s_rule, elseClauseSyntax);
```

Comment:
```suggestion
                context.ReportDiagnostic(s_rule, elseClauseSyntax.ElseKeyword);
```

This way it will report only on the else keyword instead of the whole else block.

Reply:
I made the change to the code, but the tests still succeeded without any adjustment. 😕

Reply:
They do, but I prefer you change them to validate that the analyzer only reports the else keyword instead of the else block

Reply:
Done